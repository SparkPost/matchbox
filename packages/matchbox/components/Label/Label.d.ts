export default Label;
declare function Label(props: any): any;
declare namespace Label {
    const displayName: string;
    namespace propTypes {
        const label: {
            (props: any, propName: any, componentName: any, ...rest: any[]): any;
            isRequired: any;
        };
        const children: any;
        const id: any;
        const htmlFor: any;
    }
}
