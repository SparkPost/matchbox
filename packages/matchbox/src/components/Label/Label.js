import React from 'react';
import classnames from 'classnames';
import styles from './Label.module.scss';

const Label = ({
  label,
  id,
  className,
  children
}) => (
  <label
    id={id && `${id}Label`}
    htmlFor={id}
    className={classnames(styles.Label, className)} >
    {label}
    {children}
  </label>
);

Label.displayName = 'Label';
export default Label;
