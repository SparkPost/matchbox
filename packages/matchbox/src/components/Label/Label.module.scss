@import '../../styles/config';

.Label {
  display: block;
  margin-bottom: spacing(small);
  font-weight: 400;
  font-size: font-size(300);
  line-height: rem(15);
  color: color(gray, 2);
  user-select: none;

  &:hover {
    cursor: pointer;
  }
};
