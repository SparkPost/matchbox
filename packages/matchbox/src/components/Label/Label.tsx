import React from 'react';
import { Box } from '../Box';
import { ScreenReaderOnly } from '../ScreenReaderOnly';

type LabelProps = Pick<React.ComponentProps<typeof Box>, 'fontWeight' | 'mb' | 'as'> & {
  /**
   * @deprecated Use children instead
   */
  label?: React.ReactNode;
  children?: React.ReactNode;
  id?: string;
  htmlFor?: string;
  className?: string;
  labelHidden?: boolean;
};

function Label(props: LabelProps): JSX.Element {
  const {
    label,
    id,
    htmlFor,
    className,
    children,
    labelHidden,
    as = 'label',
    fontWeight = '500',
    mb = '100',
  } = props;

  if (labelHidden) {
    return (
      <ScreenReaderOnly>
        <Box as={as} id={id && `${id}Label`} htmlFor={id}>
          {label}
          {children}
        </Box>
      </ScreenReaderOnly>
    );
  }

  return (
    <Box
      display="block"
      as={as}
      id={id && !htmlFor ? `${id}Label` : id}
      htmlFor={htmlFor ? htmlFor : id}
      fontWeight={fontWeight}
      className={className}
      mb={mb}
    >
      <Box as="div" lineHeight="200" fontSize="200">
        {label ? (
          <Box as="span" mr="200">
            {label}
          </Box>
        ) : null}
        {children ? <span>{children}</span> : null}
      </Box>
    </Box>
  );
}

Label.displayName = 'Label';

export default Label;
