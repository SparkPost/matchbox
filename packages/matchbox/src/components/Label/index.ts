export { default as Label } from './Label';
export * from './Label';
