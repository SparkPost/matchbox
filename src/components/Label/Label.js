import React, { Component } from 'react';
import styles from './Label.module.scss';

const Label = ({
  label,
  id,
}) => {
  return (
    <label
      id={id && `${id}Label`}
      htmlFor={id}
      className={styles.Label}>
      { label }
    </label>
  );
};

export default Label;
