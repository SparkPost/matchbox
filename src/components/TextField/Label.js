import React, { Component } from 'react';
import styles from './TextField.module.scss';

const Label = ({
  label,
  id,
  labelHidden
}) => (
  <label id={`${id}Label`} htmlFor='' className={classnames(labelHidden && styles.labelHidden)}>
    { label }
  </label>
);

export default Label;
