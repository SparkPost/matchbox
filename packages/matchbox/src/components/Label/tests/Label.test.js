import React from 'react';
import Label from '../Label';
import { shallow } from 'enzyme';

describe('Label', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      id: 'label1',
      label: 'Label text',
      className: 'form-field'
    };

    wrapper = shallow(<Label {...props}><span>Select one!</span></Label>);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
