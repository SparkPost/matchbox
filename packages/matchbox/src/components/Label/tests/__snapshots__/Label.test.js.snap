// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<label
  className="Label form-field"
  htmlFor="label1"
  id="label1Label"
>
  Label text
  <span>
    Select one!
  </span>
</label>
`;
